src/passes/rejectUnsupportedFeatures.ts

```diff
@@ -174,6 +174,31 @@ export class RejectUnsupportedFeatures extends ASTMapper {
     } else if (node.kind === FunctionKind.Receive) {
       throw new WillNotSupportError(`Receive functions are not supported`, node);
     }
+
+    //Checking if the function defination is constructor and
+    // if it is, checking if 'this' keyword is used to call
+    // the external/public functions of the contract
+
+    if (node.kind === FunctionKind.Constructor) {
+      const nodesWithThisIdentifier = node.vBody
+        ?.getChildren()
+        .filter((childNode, _) => childNode instanceof Identifier && childNode.raw.name === 'this');
+
+      nodesWithThisIdentifier?.forEach((identifierNode: ASTNode) => {
+        const parentNode = identifierNode?.parent;
+        if (
+          parentNode instanceof MemberAccess &&
+          parentNode?.parent instanceof FunctionCall &&
+          isExternalCall(parentNode.parent)
+        ) {
+          throw new WillNotSupportError(
+            `external function calls using this keyword not supported during contract deployment`,
```

Comment:
Please change this => "this"

Reply:
I think it should also be slightly modified as : 

'external function calls using "this" keyword not supported in contract's constructor function'
What's your say on this?